### ISSUE TITLE
Exception stack trace does not have line numbers for Self-Contained(Single Executable) apps with embedded PDBs

### ISSUE THREAD

**Issue by GF-Huang** [dotnet/runtime#39987]
### Description
The publish app as Self-Contained(Single Executable), its exception stack trace does not have line numbers, even if there is a `pdb` file at the same folder to the `exe`.

### Configuration

```
.NET Core SDK（反映任何 global.json）:
 Version:   3.1.302
 Commit:    41faccf259

运行时环境:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.302\

Host (useful for support):
  Version: 3.1.6
  Commit:  3acd9b0cd1

.NET Core SDKs installed:
  2.2.207 [C:\Program Files\dotnet\sdk]
  3.1.302 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

### Regression?

.net core 3.1.

### Other information

![image](https://user-images.githubusercontent.com/4510984/88615428-7eef4980-d0c4-11ea-965a-d86603b20ee5.png)

![image](https://user-images.githubusercontent.com/4510984/88620828-82d59880-d0d1-11ea-80e0-591a9176fb81.png)

**Comment by agocke** [comment]
I believe placing the PDBs next to the single-file is not sufficient in .NET Core 3.x. I think you need to actually embed the PDBs in the single-file application, either by using the compiler to embed the PDBs with the `<DebugType>embedded</DebugType>` option in MSBuild, or setting `<IncludeSymbolsInSingleFile>true</IncludeSymbolsInSingleFile>` during publish.

**Comment by josalem** [comment]
CC @mikem8361 @tommcdon @hoyosjs

**Comment by mikem8361** [comment]
I can't seem to repro your issue with my simple test case that throws an unhandled exception.  I made it a self-contained app and the pdb is next to the exe and I get source/line info.  Using the exact same SDK version. There may be something unique about your environment or app.  Have you tried the above suggestions (embeded PDB, etc.)?

```
SymbolTestApp starting c:\sd\builds\SymbolTestDll2\bin\Debug\netstandard2.0\win-x64
Unhandled exception. System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.Exception: Exception of type 'System.Exception' was thrown.
   at SymbolTestDll.TestClass.Foo7() in c:\sd\builds\SymbolTestDll2\TestClass.cs:line 30
   at SymbolTestDll.TestClass.Foo6(Int32 x) in c:\sd\builds\SymbolTestDll2\TestClass.cs:line 22
   at SymbolTestDll.TestClass.Foo5(Int32 x) in c:\sd\builds\SymbolTestDll2\TestClass.cs:line 17
   at SymbolTestDll.TestClass.ThrowException() in c:\sd\builds\SymbolTestDll2\TestClass.cs:line 12
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at SymbolTestApp.Program.Foo4(String dllPath) in c:\sd\builds\SymbolTestApp2\Program.cs:line 56
   at SymbolTestApp.Program.Foo2(Int32 x, String dllPath) in c:\sd\builds\SymbolTestApp2\Program.cs:line 30
   at SymbolTestApp.Program.Foo1(Int32 x, String dllPath) in c:\sd\builds\SymbolTestApp2\Program.cs:line 25
   at SymbolTestApp.Program.Main(String[] args) in c:\sd\builds\SymbolTestApp2\Program.cs:line 20
```

**Comment by GF-Huang** [comment]
@mikem8361 I will try it later and report it, thanks for reminding.

**Comment by GF-Huang** [comment]
> I believe placing the PDBs next to the single-file is not sufficient in .NET Core 3.x. I think you need to actually embed the PDBs in the single-file application, either by using the compiler to embed the PDBs with the `<DebugType>embedded</DebugType>` option in MSBuild, or setting `<IncludeSymbolsInSingleFile>true</IncludeSymbolsInSingleFile>` during publish.

Is seems not work, still not line numbers.

![image](https://user-images.githubusercontent.com/4510984/91790275-f0319780-ec42-11ea-8e43-9e0217f63996.png)

**Comment by GF-Huang** [comment]
`<DebugType>embedded</DebugType>`  works well, but why `<IncludeSymbolsInSingleFile>true</IncludeSymbolsInSingleFile>` not work?

**Comment by agocke** [comment]
We recently removed IncludeSymbolsInSingleFile from 5.0 in preference of `<DebugType>embedded`.

**Comment by mikem8361** [comment]
It sounds like this issue is resolved by adding <DebugType>embedded</DebugType> to the project file.  @agocke are we planning to support portable/full PDBs for single-contained apps?  Otherwise, we should close this issue.

**Comment by trebor678** [comment]
This still isn't working with .NET 5.0 when you produce a single file, even with `<DebugType>embedded</DebugType>` specified

**Comment by agocke** [comment]
@mikem8361 I tried this with DebugType=embedded and it looks like the line numbers are still missing in net5. I will try net6 to see if this is fixed.

**Comment by agocke** [comment]
Doesn't look like it's fixed in net6.0 either -- this looks like a bug in the debugger.

**Comment by splitice** [comment]
Can also confirm that embedded debug symbols doesnt work with .net core 5 on linux

**Comment by pspgit** [comment]
I'm also reproducing it on ASP .NET 5.0 on Windows:
- Deployment Mode: Self-contained
- File Publish Options: **Produce single file**
- DebugType: reproduces for all options: embedded, full, portable
- No line numbers

When unchecking "Produce single file" - line numbers come back.

**Comment by splitice** [comment]
Sorry I can't as none are open source. But I can say that I'm yet to see any project that does have exception lime numbers. I've been involved in 3 projects all containerised and on Linux.

**Comment by mikem8361** [comment]
@GF-Huang, does this repro with the 6.0 SDK for you? If not I would like to close this issue.

**Comment by GF-Huang** [comment]
Sorry, I haven't been paying attention to this issue lately, it's better to ask other people with similar questions to make sure .NET 6 has addressed their similar needs.

**Comment by mikem8361** [comment]
I'm assuming this is fixed in 6.0
